### ISSUE TITLE
R no longer working with build (48)

### ISSUE THREAD

**Issue by vqv** [tbodt/ish#329]
R was working in build (47), but since updating to build (48) it no longer works.  I get the following message and then dumped back to the shell.

```
Vinces-iPad-Pro-2:~# R                                                                                                                                                                    
OpenBLAS WARNING - could not determine the L2 cache size on this system, assuming 256k                                                                                                    
Error: promise already under evaluation: recursive default argument reference or earlier problems?                                                                                        
Error: promise already under evaluation: recursive default argument reference or earlier problems?                                                                                        
                                                                                                                                                                                          
R version 3.5.1 (2018-07-02) -- "Feather Spray"                                                                                                                                           
Copyright (C) 2018 The R Foundation for Statistical Computing                                                                                                                             
Platform: i686-pc-linux-gnu (32-bit)                                                                                                                                                      
                                                                                                                                                                                          
R is free software and comes with ABSOLUTELY NO WARRANTY.                                                                                                                                 
You are welcome to redistribute it under certain conditions.                                                                                                                              
Type 'license()' or 'licence()' for distribution details.                                                                                                                                 
                                                                                                                                                                                          
R is a collaborative project with many contributors.                                                                                                                                      
Type 'contributors()' for more information and                                                                                                                                            
'citation()' on how to cite R or R packages in publications.                                                                                                                              
                                                                                                                                                                                          
Type 'demo()' for some demos, 'help()' for on-line help, or                                                                                                                               
'help.start()' for an HTML browser interface to help.                                                                                                                                     
Type 'q()' to quit R.                                                                                                                                                                     
                                                                                                                                                                                          
Error: promise already under evaluation: recursive default argument reference or earlier problems?                                                                                        
Error: promise already under evaluation: recursive default argument reference or earlier problems?                                                                                        
Error: promise already under evaluation: recursive default argument reference or earlier problems?                                                                                        
Fatal error: unable to initialize the JIT
```

**Comment by jonathan-w-king** [comment]
Weird. I cannot replicate your issue...but version of R is 3.5.0:
``` Babar:/# R                                                                                           
                                                                                                     
R version 3.5.0 (2018-04-23) -- "Joy in Playing"                                                     
Copyright (C) 2018 The R Foundation for Statistical Computing                                        
Platform: i686-pc-linux-gnu (32-bit)                                                                 
                                                                                                     
R is free software and comes with ABSOLUTELY NO WARRANTY.                                            
You are welcome to redistribute it under certain conditions.                                         
Type 'license()' or 'licence()' for distribution details.                                            
                                                                                                     
R is a collaborative project with many contributors.                                                 
Type 'contributors()' for more information and                                                       
'citation()' on how to cite R or R packages in publications.                                         
                                                                                                     
Type 'demo()' for some demos, 'help()' for on-line help, or                                          
'help.start()' for an HTML browser interface to help.                                                
Type 'q()' to quit R.                                                                                
                                                                                                     
>
```
This is the version that `app add` gave me, and, yes, I did just run `ask update`, and that’s still the version I’m getting.

**Comment by tbodt** [comment]
@jonathan-w-king If you edit /etc/apk/repositories and change 3.8 to 3.9, then run apk upgrade, it'll upgrade to 3.5.1. I suspect that version will have the problem.

**Comment by jonathan-w-king** [comment]
> @jonathan-w-king If you edit /etc/apk/repositories and change 3.8 to 3.9, then run apk upgrade, it'll upgrade to 3.5.1. I suspect that version will have the problem.

Ah. So if I were a good tester, I would do that, and see if anything else breaks.

HMB for a second...and I still don’t have 3.5.1, although I do have 3.9, and R is in my world:
```
Babar:/etc# more apk/repositories                                                                    
http://dl-cdn.alpinelinux.org/alpine/v3.9/main                                                       
http://dl-cdn.alpinelinux.org/alpine/v3.9/community                                                  
Babar:/etc# more apk/world                                                                           
R                                                                                                    
R-dev                                                                                                
R-doc                                  
```
I can do `apk add R` and it says it updates, but still stuck on the (working) 3.5.0.

Which thing said, I’m not an alpine person at all, and if more is involved, I’m not doing it.

**Comment by tbodt** [comment]
Weird, alpine 3.9 should definitely have R 3.5.1...did you definitely run apk upgrade?

https://pkgs.alpinelinux.org/packages?name=R&branch=v3.9

**Comment by jonathan-w-king** [comment]
> @jonathan-w-king If you edit /etc/apk/repositories and change 3.8 to 3.9, then run apk upgrade, it'll upgrade to 3.5.1. I suspect that version will have the problem.

Ah. So if I were a good tester, I would do that, and see if anything else breaks.

HMB for a second...



> Weird, alpine 3.9 should have R 3.5.1...did you definitely run apk upgrade?
> 
> https://pkgs.alpinelinux.org/packages?name=R&branch=v3.9

Not sure how this didn’t happen before, but *now* when I ran `apk upgrade` I get updated with respect to all of the things, and R is now R 3.5.1 and goes boom on start-up with the exact same errors reported.

So...yay? 

In other news, Emacs is messed up in what I think is the same way as before (display of the buffer you are typing in goes south if the current line is more than 6 characters including 2 spaces, unless they are initial spaces...but don’t work on this until I can fully capture the issue, since it’s pretty weird).

**Comment by tbodt** [comment]
I'm gonna look into fixing this now. The emacs issue is here: #254

**Comment by jonathan-w-king** [comment]
Yup; #254 seems to be the issue I’m seeing. Anything useful for me to do there that doesn’t require a deep dive into xterm code?

**Comment by tbodt** [comment]
Not really, keep pressing C-l I guess.

**Comment by jonathan-w-king** [comment]
Yeah, so that “fixes” things in the sense that if you can still engage your ^l tick from the 80s then you won’t be harmed.

I guess what I meant is there anything I could help pin down either here or in something else likely to tickle the same display issue?

**Comment by tbodt** [comment]
Not really; it's easy to reproduce so fixing it would just require a lot of spelunking.

**Comment by tbodt** [comment]
R works fine on simulator aside from a warning about L2 cache which is just a warning.

**Comment by tbodt** [comment]
Seems to be fixed now that I fixed #326
